cpp/src/arrow/dataset/file_parquet.h

```diff
@@ -352,15 +352,17 @@ class ARROW_DS_EXPORT ParquetDatasetFactory : public DatasetFactory {
       std::shared_ptr<parquet::arrow::SchemaManifest> manifest,
       std::shared_ptr<Schema> physical_schema, std::string base_path,
       ParquetFactoryOptions options,
-      std::unordered_map<std::string, std::vector<int>> path_to_row_group_ids)
+      std::unordered_map<std::string, std::vector<int>> path_to_row_group_ids,
+      std::vector<std::string> ordered_paths)
```

Comment:
Why not take a single `std::vector<std::string, std::vector<int>> path_to_row_groups_ids` instead?

Reply:
It now takes `std::vector<std::pair<std::string, std::vector<int>>>`